src/Explorer/Controls/Settings/SettingsUtils.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { collection, container } from "./TestUtils";
+import {
+  getMaxRUs,
+  getMinRUs,
+  hasDatabaseSharedThroughput,
+  isDirty,
+  isDirtyTypes,
+  parseConflictResolutionMode,
+  parseConflictResolutionProcedure
+} from "./SettingsUtils";
+import * as DataModels from "../../../Contracts/DataModels";
+import * as ViewModels from "../../../Contracts/ViewModels";
+import ko from "knockout";
+
+describe("SettingsUtils", () => {
+  it("getMaxRUs", () => {
+    expect(collection.offer().content.collectionThroughputInfo.numPhysicalPartitions).toEqual(4);
+    expect(getMaxRUs(collection, container)).toEqual(40000);
+  });
+
+  it("getMinRUs", () => {
+    expect(collection.offer().content.collectionThroughputInfo.numPhysicalPartitions).toEqual(4);
+    expect(getMinRUs(collection, container)).toEqual(6000);
+  });
+
+  it("hasDatabaseSharedThroughput", () => {
+    expect(hasDatabaseSharedThroughput(collection)).toEqual(undefined);
+
+    const newCollection = { ...collection };
+    newCollection.getDatabase = () => {
+      return {
+        nodeKind: undefined,
+        rid: undefined,
+        container: undefined,
+        self: "",
+        id: ko.observable(""),
+        collections: ko.observableArray(undefined),
+        offer: ko.observable(undefined),
+        isDatabaseExpanded: ko.observable(false),
+        isDatabaseShared: ko.computed(() => true),
+        selectedSubnodeKind: ko.observable(undefined),
+        selectDatabase: undefined,
+        expandDatabase: undefined,
+        collapseDatabase: undefined,
+        loadCollections: undefined,
+        findCollectionWithId: undefined,
+        openAddCollection: undefined,
+        onDeleteDatabaseContextMenuClick: undefined,
+        readSettings: undefined,
+        onSettingsClick: undefined,
+        loadOffer: undefined
+      } as ViewModels.Database;
+    };
+    newCollection.offer(undefined);
+    expect(hasDatabaseSharedThroughput(newCollection)).toEqual(true);
+  });
+
+  it("parseConflictResolutionMode", () => {
+    expect(parseConflictResolutionMode("custom")).toEqual(DataModels.ConflictResolutionMode.Custom);
+    expect(parseConflictResolutionMode("lastwriterwins")).toEqual(DataModels.ConflictResolutionMode.LastWriterWins);
+  });
+
+  it("parseConflictResolutionProcedure", () => {
+    expect(parseConflictResolutionProcedure("/dbs/db/colls/coll/sprocs/conflictResSproc")).toEqual("conflictResSproc");
+    expect(parseConflictResolutionProcedure("conflictResSproc")).toEqual("conflictResSproc");
+  });
+
+  describe("isDirty", () => {
```

Comment:
Jest.each would be a good fit for these tests https://jestjs.io/blog/2018/05/29/jest-23-blazing-fast-delightful-testing#jest-each

Here is an example https://github.com/Azure/cosmos-explorer/blob/401ee7af2cb61c1f012cfe21d637f416e9ce843d/src/Utils/MessageValidation.test.ts